internal/policy/container/has_modified_files.go

```diff
@@ -212,35 +252,61 @@ func untar(pathChan chan<- string, r io.Reader) error {
 
 		// check the file type
 		switch header.Typeflag {
-		// if its a dir ignore it
+		// if its a dir and it doesn't exist create it
 		case tar.TypeDir:
-			continue
+			if err := fs.MkdirAll(target, 0o755); err != nil {
+				return err
+			}
 
-		// if it's a file write the name to the channel
+		// if it's a file send path and create it if in RPM db
 		case tar.TypeReg:
 			// Strip off any leading '/' or './'
 			pathChan <- strings.TrimLeft(target, "./")
 
-		// if it's a link create it
+			// All we care about to write to the fs is RPM db.
+			if !strings.HasPrefix(strings.TrimLeft(target, "./"), "var/lib/rpm") {
+				continue
+			}
+
+			if err := func() error {
+				f, err := fs.OpenFile(target, os.O_CREATE|os.O_RDWR, os.FileMode(header.Mode))
+				if err != nil {
+					return err
+				}
+				defer f.Close()
+
+				// copy over contents
+				if _, err := io.Copy(f, tr); err != nil {
+					return err
+				}
+
+				return nil
+			}(); err != nil {
+				return err
+			}
+
+		// if it's a link send path
 		case tar.TypeSymlink:
+			// Strip off any leading '/' or './'
 			pathChan <- strings.TrimLeft(header.Linkname, "./")
 		}
 	}
 }
 
-// getInstalledFilesFor returns a map of installed files by pkgs. The returned map only contains
-// relevant keys to aid in lookup; values are unimportant.
-func (p *HasModifiedFilesCheck) getInstalledFilesFor(pkgList []*rpmdb.PackageInfo) (map[string]struct{}, error) {
-	installedFiles := make(map[string]struct{}, len(pkgList))
+// getInstalledFilesFor returns a map of installed files by pkgs. The returned map gives
+// the file as the key, and packageName-version as the value.
+func (p *HasModifiedFilesCheck) getInstalledFilesFor(pkgList []*rpmdb.PackageInfo) (map[string]string, error) {
+	installedFiles := make(map[string]string, len(pkgList))
 	for _, pkg := range pkgList {
-		filenames, err := pkg.InstalledFileNames()
+		filenames, err := pkg.InstalledFiles()
 		if err != nil {
 			return nil, err
 		}
 		for _, file := range filenames {
 			// A struct is used here, but it is unimportant and
 			// should not have value.
-			installedFiles[file] = struct{}{}
+			// Also trims any leading "." or "/"
+			installedFiles[strings.TrimLeft(file.Path, "./")] = fmt.Sprintf("%s-%s", pkg.Name, pkg.Version)
```

Comment:
Is this comment valid anymore? its not a `map[string]struct{}` anymore it's a `map[string]string` now, so should the comment be `this is a string that is trimmed`?

Reply:
I'll modify it.